plugins/nf-tower/src/main/io/seqera/tower/plugin/TowerFusionEnv.groovy

```diff
@@ -0,0 +1,257 @@
+
+    // Default connection timeout for HTTP requests
+    private static final Duration DEFAULT_CONNECTION_TIMEOUT = Duration.of(30, ChronoUnit.SECONDS)
+
+    // Default retry policy settings for HTTP requests: delay, max delay, attempts, and jitter
+    private static final Duration DEFAULT_RETRY_POLICY_DELAY = Duration.of(450, ChronoUnit.MILLIS)
+    private static final Duration DEFAULT_RETRY_POLICY_MAX_DELAY = Duration.of(90, ChronoUnit.SECONDS)
+    private static final int DEFAULT_RETRY_POLICY_MAX_ATTEMPTS = 10
+    private static final double DEFAULT_RETRY_POLICY_JITTER = 0.5
+
+    // The HttpClient instance used to send requests
+    private final HttpClient httpClient = newDefaultHttpClient()
+
+    // The RetryPolicy instance used to retry requests
+    private final RetryPolicy retryPolicy = newDefaultRetryPolicy(SERVER_ERRORS)
+
+    // Nextflow session
+    private final Session session
+
+    // Platform endpoint to use for requests
+    private final String endpoint
+
+    // Platform access token to use for requests
+    private final String accessToken
+
+    /**
+     * Constructor for the class. It initializes the session, endpoint, and access token.
+     */
+    TowerFusionEnv() {
+        this.session = Global.session as Session
+        final towerConfig = session.config.navigate('tower') as Map ?: [:]
+        final env = SysEnv.get()
+        this.endpoint = PlatformHelper.getEndpoint(towerConfig, env)
+        this.accessToken = PlatformHelper.getAccessToken(towerConfig, env)
+    }
+
+    /**
+     * Return any environment variables relevant to Fusion execution. This method is called
+     * by {@link nextflow.fusion.FusionEnvProvider#getEnvironment} to determine which
+     * environment variables are needed for the current run.
+     *
+     * @param scheme The scheme for which the environment variables are needed (currently unused)
+     * @param config The Fusion configuration object
+     * @return A map of environment variables
+     */
+    @Override
+    Map<String, String> getEnvironment(String scheme, FusionConfig config) {
+
+        // TODO(amiranda): Hardcoded for now. We need to find out how to obtain
+        // the concrete product SKU and version. Candidate: FusionConfig?
+        final product = 'fusion'
+        final version = '2.4'
+
+        try {
+            final token = getLicenseToken(product, version)
+            return [
+                FUSION_LICENSE_TOKEN: token,
+            ]
+        } catch (Exception e) {
+            log.warn("Error retrieving Fusion license information: ${e.message}")
+            return Map.of()
+        }
+    }
+
+    /**
+     * Send a request to Platform to obtain a license-scoped JWT for Fusion. The request is authenticated using the
+     * Platform access token provided in the configuration of the current session.
+     *
+     * @throws AbortOperationException if a Platform access token cannot be found
```

Comment:
It also throws other exceptions if the token is not valid or malformed

Reply:
Addressed with 1fed85918c016495c598fc805f4bbce4d1d3fce4 and refactored into a9849f55891aebac3bb04a4a75b98e18ad39e5a5